### PR TITLE
Fix regression introduced by types lying about undefined vs null.

### PR DESCRIPTION
**Pull request by ppershing**
Note that this is just a hotfix and the whole typing will be refactored in the future.

### CODE REVIEW

File: src/cardano.ts [hunk 1]
```diff
@@ -606,7 +606,7 @@ export function serializePoolRelayParams(relayParams: RelayParams): Buffer {
   noBuf.writeUInt8(RELAY_NO);
 
   let portBuf: Buffer;
-  if ('portNumber' in params) {
+  if ('portNumber' in params && params.portNumber) {
```
Comment (refi93) [inline src/cardano.ts:609]:
```suggestion
  if ('portNumber' in params && params.portNumber != null) {
```

File: src/cardano.ts [hunk 2]
```diff
@@ -670,7 +670,7 @@ export function serializePoolRelayParams(relayParams: RelayParams): Buffer {
   }
 
   let dnsBuf: Buffer | undefined;
-  if ('dnsName' in params) {
+  if ("dnsName" in params && params.dnsName) {
```
Comment (refi93) [inline src/cardano.ts:673]:
```suggestion
  if ("dnsName" in params && params.dnsName != null) {
```

File: src/cardano.ts (hunk 1 above)
Comment (refi93) [reply src/cardano.ts:609]: technically, portnumber can be 0

File: src/cardano.ts (hunk 2 above)
Comment (refi93) [reply src/cardano.ts:673]: teeechnically, dns name can be empty string

File: src/cardano.ts [hunk 3]
```diff
@@ -205,7 +205,7 @@ export function validateTransaction(
     // we try to serialize the data, an error is thrown if ada amount or address params are invalid
     serializeOutputBasicParams(output, protocolMagic, networkId);
 
-    if ('spendingPath' in output) {
+    if ("spendingPath" in output && output.spendingPath != null) {
```
Comment (refi93) [inline src/cardano.ts:208]: is there any point in checking whether given property is in the object if we are doing the `!= null` comparison? I know this is a hotfix, just asking if I'm perhaps missing something

File: src/cardano.ts (hunk 1 above)
Comment (refi93) [reply src/cardano.ts:609]: indeed, I noticed the original code has the same problem

**Review by refi93 — APPROVED** [review]

File: src/cardano.ts (hunk 3 above)
Comment (ppershing) [reply src/cardano.ts:208]: The check is covered, but Typescript does not like checks on maybe-undefined properties (especially the ones coming from union types)

File: src/cardano.ts (hunk 2 above)
Comment (ppershing) [reply src/cardano.ts:673]: Yep. This condition allows for empty dns name to pass. We can deal with the superexact semantic (can be empty) later

**Pull request merged**